test/evp_kdf_test.c

```diff
@@ -27,26 +27,37 @@ static EVP_KDF_CTX *get_kdfbyname(const char *name)
     return kctx;
 }
 
+static OSSL_PARAM *construct_tls1_prf_params(const char *digest, const char *secret,
+    const char *seed)
+{
+    OSSL_PARAM *params = OPENSSL_malloc(sizeof(OSSL_PARAM) * 4);
+    OSSL_PARAM *p = params;
+
+    *p++ = OSSL_PARAM_construct_utf8_string(OSSL_KDF_PARAM_DIGEST,
+                                            (char *)digest, strlen(digest));
```

Comment:
When constructing utf8 strings, just pass 0 for the length.  The function does the `strlen(3)` call for you.  I've toyed with removing the possibility a few times but there are occasional efficiency gains when the length is known.

An aside: the length was being set very inconsistently either as `strlen(3)` or `strlen(3) + 1`.

Reply:
Ok, fixed now.